migrations/20210612_initial.sql

```diff
@@ -0,0 +1,104 @@
+CREATE TABLE profiles (
+  profile_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL,
+  properties TEXT NOT NULL -- JSON with user or contact profile
+);
+
+CREATE TABLE known_servers(
+  host TEXT NOT NULL,
+  port TEXT NOT NULL,
+  key_hash BLOB,
+  PRIMARY KEY (host, port)
+) WITHOUT ROWID;
+
+CREATE TABLE contacts (
+  contact_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL, -- display name set by local user
+  properties TEXT NOT NULL, -- JSON set by local user
+  profile_id INTEGER REFERENCES profiles, -- sent by remote contact
+  contact_status TEXT NOT NULL DEFAULT ''
+);
+
+CREATE TABLE contact_connections (
+  connection_id BLOB NOT NULL PRIMARY KEY,
+  contact_id INTEGER NOT NULL REFERENCES contacts ON DELETE RESTRICT,
+  conn_level INTEGER NOT NULL DEFAULT 0,
```

Comment:
I'd remove default so that one has to remember to calculate and set `conn_level` from the app, otherwise you risk forgetting to calculate a level different from zero and finding out too late. There's no downside to let this constraint fail - it would only reveal an error in implementation, it doesn't pose a risk to breaking end user's application. As specified by `NOT NULL`, this column is part of the logic and always has to be set, it's not just some descriptive optional field, if anything having `DEFAULT 0` for it only water the constraint's meaning down - why is it specifically 0 that is default, why not 1 or 2 - they would have the same semantic weight - setting a 0 for level 2 conn would be as much a mistake as setting a 2 for level 0 conn